process/sync/baseSync.go

```diff
@@ -826,6 +828,24 @@ func (boot *baseBootstrap) rollBack(revertUsingForkNonce bool) error {
 	return nil
 }
 
+func (boot *baseBootstrap) shouldAllowRollback(currHeader data.HeaderHandler) bool {
+	finalBlockNonce := boot.forkDetector.GetHighestFinalBlockNonce()
+	isRollBackBehindFinal := currHeader.GetNonce() <= finalBlockNonce
+	isFinalBlockRollBack := currHeader.GetNonce() == finalBlockNonce
```

Comment:
so this will ensure that we only revert once because the highest final block nonce in the fork detector will keep the last value.
What happens if the node is slightly behind and the fork detector has a nonce slightly larger than one? The rollback will never happen and the node will still be stuck.

Reply:
This scenario should not be happen, as there is no reason to roll back two blocks behind final. Fork detector never returns a final block nonce higher than your last processed and committed even if the tracker feeds it with the future blocks from the longest chain. Fork detector returns the minimum nonce between the higher received as notarized by meta (from tracker) and the higher processed and committed which match with the one added by the tracker in the longest chain.